unyt/array.py

```diff
@@ -982,8 +975,6 @@ def in_base(self, unit_system=None):
         else:
             to_units = self.units.get_base_equivalent(unit_system)
             conv, offset = self.units.get_conversion_factor(to_units, self.dtype)
-        new_dtype = np.dtype("f" + str(self.dtype.itemsize))
```

Comment:
Why doesn't this require the same logic as above, instead of just getting rid of these two lines?

Reply:
Because `in_base` doesn't explicitly create a new array like `in_units` does. The conversion factor `conv` has the correct dtype and numpy casting rules produce the desired result.